contracts/mediator/BasicMediator.sol

```diff
@@ -0,0 +1,121 @@
+pragma solidity 0.4.24;
+
+import "./Initializable.sol";
+import "./AMBMediator.sol";
+import "./ERC721Bridge.sol";
+import "./Claimable.sol";
+import "./Upgradeable.sol";
+
+contract BasicMediator is Initializable, AMBMediator, ERC721Bridge, Upgradeable, Claimable {
+    event FailedMessageFixed(bytes32 indexed dataHash, address recipient, uint256 tokenId);
+
+    bytes4 internal constant GET_KITTY = 0xe98b7f4d; // getKitty(uint256)
+
+    function initialize(
+        address _bridgeContract,
+        address _mediatorContract,
+        address _erc721token,
+        uint256 _requestGasLimit,
+        address _owner
+    ) external returns (bool) {
+        require(!isInitialized());
+
+        uintStorage[DEPLOYED_AT_BLOCK] = block.number;
+        _setBridgeContract(_bridgeContract);
+        _setMediatorContractOnOtherSide(_mediatorContract);
+        setErc721token(_erc721token);
+        _setRequestGasLimit(_requestGasLimit);
+        setOwner(_owner);
+        setInitialize();
+
+        return isInitialized();
+    }
+
+    function getBridgeInterfacesVersion() external pure returns (uint64 major, uint64 minor, uint64 patch) {
+        return (1, 0, 0);
+    }
+
+    function getBridgeMode() external pure returns (bytes4 _data) {
+        return bytes4(keccak256(abi.encodePacked("nft-to-nft-amb")));
+    }
+
+    function transferToken(uint256 _tokenId) external {
+        ERC721 token = erc721token();
+        address from = msg.sender;
+        address to = address(this);
+
+        token.transferFrom(from, to, _tokenId);
+        bridgeSpecificActionsOnTokenTransfer(from, _tokenId);
+    }
+
+    /**
+    *  getKitty(uint256) returns:
+    *       bool isGestating,
+    *       bool isReady,
+    *       uint256 cooldownIndex,
+    *       uint256 nextActionAt,
+    *       uint256 siringWithId,
+    *       uint256 birthTime,
+    *       uint256 matronId,
+    *       uint256 sireId,
+    *       uint256 generation,
+    *       uint256 genes
+    **/
+    function getMetadata(uint256 _tokenId) internal view returns (bytes memory metadata) {
+        bytes memory callData = abi.encodeWithSelector(GET_KITTY, _tokenId);
+        address tokenAddress = erc721token();
+        metadata = new bytes(320);
+        assembly {
```

Comment:
I had to use `assembly` to get metadata and store it as `bytes` because of `Stack Too Deep` error produced by the large amount of variable needed to handle the result of the metadata along with the logic of other methods.